src/search3_httpd.erl

```diff
@@ -72,6 +74,13 @@ handle_analyze_req(#httpd{method='POST'}=Req) ->
 handle_analyze_req(Req) ->
     send_method_not_allowed(Req, "GET,POST").
 
+handle_cleanup_req(#httpd{method='POST'}=Req, Db) ->
+    Tries = config:get_integer("search3", "index_cleanup_attempts", 10),
```

Comment:
you need to check the content-type is application/json for all POST requests to couchdb.